sequence_processing_pipeline/NuQCJob.py

```diff
@@ -44,7 +44,8 @@ def __init__(self, fastq_root_dir, output_path, sample_sheet_path,
                  wall_time_limit, jmem, fastp_path, minimap2_path,
                  samtools_path, modules_to_load, qiita_job_id,
                  max_array_length, known_adapters_path, movi_path, gres_value,
-                 pmls_path, bucket_size=8, length_limit=100, cores_per_task=4):
+                 pmls_path, bucket_size=8, length_limit=100, cores_per_task=4,
+                 additional_fastq_tags=None):
```

Comment:
Why define this parameters as None/list vs. just list? I mean an empty list means that there is nothing, no?

Reply:
Maybe I am misunderstanding the question, but isn’t avoiding mutable default arguments thr preferred approach in python? https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Reply:
Yes, that was my question; thank you for the link. Do you also think is worth checking that is not None and an instance of list in the if/else below? 

FWIW in my own code I try not change any parameters in a function; just seems odd to me but I completely understand the point.

Reply:
😄 Agreed! My apologies guys, it seems like I've developed a knee-jerk habit to preserve the legacy path. I agree this looks cleaner now.